lib/file_helper.rb

```diff
@@ -49,10 +49,16 @@ def self.download(
     skip_rate_limit: false,
     verbose: false,
     validate_uri: true,
-    retain_on_max_file_size_exceeded: false
+    retain_on_max_file_size_exceeded: false,
+    use_port_for_host_header: false
   )
     url = "https:" + url if url.start_with?("//")
-    raise Discourse::InvalidParameters.new(:url) unless url =~ %r{\Ahttps?://}
+
+    # Certain system tests need to download files from http:// URLs (e.g. minio S3 tests)
+    # which are hosted on localhost.
+    if Rails.env.production?
```

Comment:
Also open to changes with this, could potentially be another option...

Reply:
The check here is `url =~ %r{\Ahttps?://}`

So the `s` is optional, and so downloading from `http://` should already be allowed? 🤔

Reply:
Hm that's odd, will try to remove it and see if it's fine; a lot of these little fixes were from when I was first experimenting so maybe it's not relevant anymore